## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 400938`

Logs: https://github.com/JohnRTitor/nixpkgs-review-gha/actions/runs/15299953450


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:fast_forward: 6 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.xone</li>
    <li>linuxKernel.packages.linux_5_10_hardened.xone</li>
    <li>linuxKernel.packages.linux_5_4.xone</li>
    <li>linuxKernel.packages.linux_5_4_hardened.xone</li>
    <li>linuxPackages_5_10_hardened.xone</li>
    <li>linuxPackages_5_4_hardened.xone</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.xone</li>
    <li>linuxPackages_5_15_hardened.xone (linuxKernel.packages.linux_5_15_hardened.xone)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 19 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_6_1.xone</li>
    <li>linuxPackages.xone (linuxKernel.packages.linux_6_12.xone)</li>
    <li>linuxPackages_hardened.xone (linuxPackages_6_12_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_13.xone</li>
    <li>linuxPackages_6_13_hardened.xone (linuxKernel.packages.linux_6_13_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_14.xone</li>
    <li>linuxPackages_6_14_hardened.xone (linuxKernel.packages.linux_6_14_hardened.xone)</li>
    <li>linuxPackages_latest.xone (linuxKernel.packages.linux_6_15.xone)</li>
    <li>linuxPackages_6_1_hardened.xone (linuxKernel.packages.linux_6_1_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_6.xone</li>
    <li>linuxPackages_6_6_hardened.xone (linuxKernel.packages.linux_6_6_hardened.xone)</li>
    <li>linuxPackages_ham.xone (linuxKernel.packages.linux_ham.xone)</li>
    <li>linuxPackages_latest-libre.xone (linuxKernel.packages.linux_latest_libre.xone)</li>
    <li>linuxPackages-libre.xone (linuxKernel.packages.linux_libre.xone)</li>
    <li>linuxPackages_lqx.xone (linuxKernel.packages.linux_lqx.xone)</li>
    <li>linuxPackages_xanmod.xone (linuxKernel.packages.linux_xanmod.xone)</li>
    <li>linuxPackages_xanmod_latest.xone (linuxKernel.packages.linux_xanmod_latest.xone, linuxPackages_xanmod_stable.xone)</li>
    <li>linuxPackages_zen.xone (linuxKernel.packages.linux_zen.xone)</li>
    <li>xow_dongle-firmware</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_15.xone</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash -C /nix/store/g65kb2myl3xm1dq9p0fgndyzwxjhf37i-linux-5.15.184-dev/lib/modules/5.15.184/build M=\$\(sourceRoot\) VERSION=0.3.1 modules
  CC [M]  /build/source/transport/wired.o
  CC [M]  /build/source/transport/dongle.o
  CC [M]  /build/source/transport/mt76.o
  CC [M]  /build/source/bus/bus.o
/build/source/transport/mt76.c: In function 'xone_mt76_mac_looks_counterfeit':
/build/source/transport/mt76.c:722:9: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  722 |         for (int i = 0; i < ARRAY_SIZE(xone_counterfeit_macs); i++) {
      |         ^~~
/build/source/transport/mt76.c:722:9: note: use option '-std=c99', '-std=gnu99', '-std=c11' or '-std=gnu11' to compile your code
make[1]: *** [/nix/store/g65kb2myl3xm1dq9p0fgndyzwxjhf37i-linux-5.15.184-dev/lib/modules/5.15.184/source/scripts/Makefile.build:289: /build/source/transport/mt76.o] Error 1
make[1]: *** Waiting for unfinished jobs....
/build/source/transport/dongle.c: In function 'xone_dongle_init':
/build/source/transport/dongle.c:892:46: warning: too many arguments for format [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-extra-args-Wformat-extra-args8;;]
  892 |                         snprintf(fwname, 25, "xow_dongle.bin",
      |                                              ^~~~~~~~~~~~~~~~
make: *** [/nix/store/g65kb2myl3xm1dq9p0fgndyzwxjhf37i-linux-5.15.184-dev/lib/modules/5.15.184/source/Makefile:1914: /build/source] Error 2</pre>
</details>
<details>
<summary>linuxPackages_5_15_hardened.xone</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/xy4jjgw87sbgwylm5kn047d9gkbhsr9x-bash-5.2p37/bin/bash -C /nix/store/q47mggckdq1aa5kh94zzwxz74g70xy18-linux-hardened-5.15.183-dev/lib/modules/5.15.183-hardened1/build M=\$\(sourceRoot\) VERSION=0.3.1 modules
  CC [M]  /build/source/transport/wired.o
  CC [M]  /build/source/transport/dongle.o
  CC [M]  /build/source/transport/mt76.o
  CC [M]  /build/source/bus/bus.o
/build/source/transport/dongle.c: In function 'xone_dongle_init':
/build/source/transport/dongle.c:892:46: warning: too many arguments for format [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-extra-args-Wformat-extra-args8;;]
  892 |                         snprintf(fwname, 25, "xow_dongle.bin",
      |                                              ^~~~~~~~~~~~~~~~
/build/source/transport/mt76.c: In function 'xone_mt76_mac_looks_counterfeit':
/build/source/transport/mt76.c:722:9: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  722 |         for (int i = 0; i < ARRAY_SIZE(xone_counterfeit_macs); i++) {
      |         ^~~
/build/source/transport/mt76.c:722:9: note: use option '-std=c99', '-std=gnu99', '-std=c11' or '-std=gnu11' to compile your code
make[1]: *** [/nix/store/q47mggckdq1aa5kh94zzwxz74g70xy18-linux-hardened-5.15.183-dev/lib/modules/5.15.183-hardened1/source/scripts/Makefile.build:289: /build/source/transport/mt76.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [/nix/store/q47mggckdq1aa5kh94zzwxz74g70xy18-linux-hardened-5.15.183-dev/lib/modules/5.15.183-hardened1/source/Makefile:1914: /build/source] Error 2</pre>
</details>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:fast_forward: 16 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.xone</li>
    <li>linuxKernel.packages.linux_5_10_hardened.xone</li>
    <li>linuxKernel.packages.linux_5_4.xone</li>
    <li>linuxKernel.packages.linux_5_4_hardened.xone</li>
    <li>linuxKernel.packages.linux_lqx.xone</li>
    <li>linuxKernel.packages.linux_xanmod.xone</li>
    <li>linuxKernel.packages.linux_xanmod_latest.xone</li>
    <li>linuxKernel.packages.linux_xanmod_stable.xone</li>
    <li>linuxKernel.packages.linux_zen.xone</li>
    <li>linuxPackages_5_10_hardened.xone</li>
    <li>linuxPackages_5_4_hardened.xone</li>
    <li>linuxPackages_lqx.xone</li>
    <li>linuxPackages_xanmod.xone</li>
    <li>linuxPackages_xanmod_latest.xone</li>
    <li>linuxPackages_xanmod_stable.xone</li>
    <li>linuxPackages_zen.xone</li>
  </ul>
</details>
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_15.xone</li>
    <li>linuxPackages_5_15_hardened.xone (linuxKernel.packages.linux_5_15_hardened.xone)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 15 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_6_1.xone</li>
    <li>linuxPackages.xone (linuxKernel.packages.linux_6_12.xone)</li>
    <li>linuxPackages_hardened.xone (linuxPackages_6_12_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_13.xone</li>
    <li>linuxPackages_6_13_hardened.xone (linuxKernel.packages.linux_6_13_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_14.xone</li>
    <li>linuxPackages_6_14_hardened.xone (linuxKernel.packages.linux_6_14_hardened.xone)</li>
    <li>linuxPackages_latest.xone (linuxKernel.packages.linux_6_15.xone)</li>
    <li>linuxPackages_6_1_hardened.xone (linuxKernel.packages.linux_6_1_hardened.xone)</li>
    <li>linuxKernel.packages.linux_6_6.xone</li>
    <li>linuxPackages_6_6_hardened.xone (linuxKernel.packages.linux_6_6_hardened.xone)</li>
    <li>linuxPackages_ham.xone (linuxKernel.packages.linux_ham.xone)</li>
    <li>linuxPackages_latest-libre.xone (linuxKernel.packages.linux_latest_libre.xone)</li>
    <li>linuxPackages-libre.xone (linuxKernel.packages.linux_libre.xone)</li>
    <li>xow_dongle-firmware</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>linuxKernel.packages.linux_5_15.xone</summary>
<pre>no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/s1qkj0ph0ma64a6743mvkwnabrbw1hsc-bash-5.2p37/bin/bash -C /nix/store/ba3xnj9b80gwr4120pp9dwmdr7sszyc4-linux-5.15.184-dev/lib/modules/5.15.184/build M=\$\(sourceRoot\) VERSION=0.3.1 modules
  CC [M]  /build/source/transport/wired.o
  CC [M]  /build/source/transport/dongle.o
  CC [M]  /build/source/transport/mt76.o
  CC [M]  /build/source/bus/bus.o
/build/source/transport/dongle.c: In function 'xone_dongle_init':
/build/source/transport/dongle.c:892:46: warning: too many arguments for format [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-extra-args-Wformat-extra-args8;;]
  892 |                         snprintf(fwname, 25, "xow_dongle.bin",
      |                                              ^~~~~~~~~~~~~~~~
/build/source/transport/mt76.c: In function 'xone_mt76_mac_looks_counterfeit':
/build/source/transport/mt76.c:722:9: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  722 |         for (int i = 0; i < ARRAY_SIZE(xone_counterfeit_macs); i++) {
      |         ^~~
/build/source/transport/mt76.c:722:9: note: use option '-std=c99', '-std=gnu99', '-std=c11' or '-std=gnu11' to compile your code
make[1]: *** [/nix/store/ba3xnj9b80gwr4120pp9dwmdr7sszyc4-linux-5.15.184-dev/lib/modules/5.15.184/source/scripts/Makefile.build:289: /build/source/transport/mt76.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [/nix/store/ba3xnj9b80gwr4120pp9dwmdr7sszyc4-linux-5.15.184-dev/lib/modules/5.15.184/source/Makefile:1914: /build/source] Error 2</pre>
</details>
<details>
<summary>linuxPackages_5_15_hardened.xone</summary>
<pre>  CC [M]  /build/source/transport/dongle.o
  CC [M]  /build/source/transport/mt76.o
  CC [M]  /build/source/bus/bus.o
  CC [M]  /build/source/bus/protocol.o
/nix/store/ammn9r8dxwx0rdssbl6y9mlgrx4g2808-binutils-2.44/bin/ld: warning: -z relro ignored
/build/source/transport/mt76.c: In function 'xone_mt76_mac_looks_counterfeit':
  CC [M]  /build/source/auth/auth.o
/build/source/transport/mt76.c:722:9: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  722 |         for (int i = 0; i < ARRAY_SIZE(xone_counterfeit_macs); i++) {
      |         ^~~
/build/source/transport/mt76.c:722:9: note: use option '-std=c99', '-std=gnu99', '-std=c11' or '-std=gnu11' to compile your code
/build/source/transport/dongle.c: In function 'xone_dongle_init':
/build/source/transport/dongle.c:892:46: warning: too many arguments for format [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wformat-extra-args-Wformat-extra-args8;;]
  892 |                         snprintf(fwname, 25, "xow_dongle.bin",
      |                                              ^~~~~~~~~~~~~~~~
make[1]: *** [/nix/store/wz1avc4qcr00jwb30vxhrd6vwamdrfyn-linux-hardened-5.15.183-dev/lib/modules/5.15.183-hardened1/source/scripts/Makefile.build:289: /build/source/transport/mt76.o] Error 1
make[1]: *** Waiting for unfinished jobs....
/nix/store/ammn9r8dxwx0rdssbl6y9mlgrx4g2808-binutils-2.44/bin/ld: warning: -z relro ignored
/nix/store/ammn9r8dxwx0rdssbl6y9mlgrx4g2808-binutils-2.44/bin/ld: warning: -z relro ignored
make: *** [/nix/store/wz1avc4qcr00jwb30vxhrd6vwamdrfyn-linux-hardened-5.15.183-dev/lib/modules/5.15.183-hardened1/source/Makefile:1914: /build/source] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/xo/xow_dongle-firmware/package.nix
+++ b/pkgs/by-name/xo/xow_dongle-firmware/package.nix
@@ -47,7 +47,10 @@ stdenvNoCC.mkDerivation rec {
     description = "Xbox One wireless dongle firmware";
     homepage = "https://www.xbox.com/en-NZ/accessories/adapters/wireless-adapter-windows";
     license = licenses.unfree;
-    maintainers = with maintainers; [ rhysmdnz ];
+    maintainers = with maintainers; [
+      rhysmdnz
+      fazzi
+    ];
     platforms = platforms.linux;
   };
 }

--- a/pkgs/by-name/xo/xow_dongle-firmware/package.nix
+++ b/pkgs/by-name/xo/xow_dongle-firmware/package.nix
@@ -4,36 +4,50 @@
   fetchurl,
   cabextract,
 }:
-
 stdenvNoCC.mkDerivation rec {
   pname = "xow_dongle-firmware";
-  version = "2017-07";
+  version = "0-unstable-2025-04-22";
+
+  srcs = [
+    (fetchurl {
+      name = "xow_dongle.cab";
+      url = "http://download.windowsupdate.com/c/msdownload/update/driver/drvs/2017/07/1cd6a87c-623f-4407-a52d-c31be49e925c_e19f60808bdcbfbd3c3df6be3e71ffc52e43261e.cab";
+      hash = "sha256-ZXNqhP9ANmRbj47GAr7ZGrY1MBnJyzIz3sq5/uwPbwQ=";
+    })
+    (fetchurl {
+      name = "xow_dongle_045e_02e6.cab";
+      url = "https://catalog.s.download.windowsupdate.com/d/msdownload/update/driver/drvs/2015/12/20810869_8ce2975a7fbaa06bcfb0d8762a6275a1cf7c1dd3.cab";
+      hash = "sha256-5jiKJ6dXVpIN5zryRo461V16/vWavDoLUICU4JHRnwg=";
+    })
+  ];
+
+  sourceRoot = ".";
 
   dontConfigure = true;
   dontBuild = true;
 
-  src = fetchurl {
-    url = "http://download.windowsupdate.com/c/msdownload/update/driver/drvs/2017/07/1cd6a87c-623f-4407-a52d-c31be49e925c_e19f60808bdcbfbd3c3df6be3e71ffc52e43261e.cab";
-    sha256 = "013g1zngxffavqrk5jy934q3bdhsv6z05ilfixdn8dj0zy26lwv5";
-  };
-
   nativeBuildInputs = [ cabextract ];
 
-  sourceRoot = ".";
+  unpackPhase = ''
+    sources=($srcs)
 
-  unpackCmd = ''
-    cabextract -F FW_ACC_00U.bin ${src}
+    cabextract -F FW_ACC_00U.bin ''${sources[0]}
+    mv FW_ACC_00U.bin xow_dongle.bin
+
+    cabextract -F FW_ACC_00U.bin ''${sources[1]}
+    mv FW_ACC_00U.bin xow_dongle_045e_02e6.bin
   '';
 
   installPhase = ''
-    install -Dm644 FW_ACC_00U.bin ${placeholder "out"}/lib/firmware/xow_dongle.bin
+    install -Dm644 xow_dongle.bin $out/lib/firmware/xow_dongle.bin
+    install -Dm644 xow_dongle_045e_02e6.bin $out/lib/firmware/xow_dongle_045e_02e6.bin
   '';
 
   meta = with lib; {
     description = "Xbox One wireless dongle firmware";
     homepage = "https://www.xbox.com/en-NZ/accessories/adapters/wireless-adapter-windows";
     license = licenses.unfree;
-    maintainers = with lib.maintainers; [ rhysmdnz ];
+    maintainers = with maintainers; [ rhysmdnz ];
     platforms = platforms.linux;
   };
 }

--- a/pkgs/os-specific/linux/xone/default.nix
+++ b/pkgs/os-specific/linux/xone/default.nix
@@ -4,16 +4,15 @@
   fetchFromGitHub,
   kernel,
 }:
-
 stdenv.mkDerivation (finalAttrs: {
   pname = "xone";
-  version = "0.3-unstable-2024-12-23";
+  version = "0.3.1";
 
   src = fetchFromGitHub {
     owner = "dlundqvist";
     repo = "xone";
-    rev = "6b9d59aed71f6de543c481c33df4705d4a590a31";
-    hash = "sha256-MpxP2cb0KEPKaarjfX/yCbkxIFTwwEwVpTMhFcis+A4=";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-qMZlQgAe5vB5zfuhyK7EBxIwfhnC5MvnF/qr3BGnDms=";
   };
 
   setSourceRoot = ''
@@ -43,6 +42,6 @@ stdenv.mkDerivation (finalAttrs: {
       fazzi
     ];
     platforms = platforms.linux;
-    broken = kernel.kernelOlder "5.11";
+    broken = kernel.kernelOlder "6";
   };
 })

